plugins/aws/iam/crossAccountAccess.js

```diff
@@ -0,0 +1,64 @@
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Cross Account Access',
+    category: 'IAM',
+    description: 'Ensures that either MFA or external IDs are used to access AWS resources.',
+    more_info: 'MFA/external ID adds an extra layer of security on top of roles, temporary security credentials and facilitates external third-party accounts to access your AWS resources in a secure way.',
+    link: 'https://aws.amazon.com/blogs/aws/mfa-protection-for-cross-account-access/',
+    recommended_action: 'Edit IAM role to require MFA or external ID.',
+    apis: ['IAM:listRoles'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        
+        var region = helpers.defaultRegion(settings);
+
+        var listRoles = helpers.addSource(cache, source,
+            ['iam', 'listRoles', region]);
+
+        if (!listRoles) return callback(null, results, source);
+
+        if (listRoles.err || !listRoles.data) {
+            helpers.addResult(results, 3,
+                'Unable to query for IAM roles: ' + helpers.addError(listRoles));
+            return callback(null, results, source);
+        }
+
+        if (!listRoles.data.length) {
+            helpers.addResult(results, 0, 'No IAM roles found');
+            return callback(null, results, source);
+        }
+
+        listRoles.data.forEach(role => {
+            if (!role.AssumeRolePolicyDocument) {
+                return;
+            }
+
+            var assumeRolePolicy = JSON.parse(decodeURIComponent(role.AssumeRolePolicyDocument));
+            
+            if (assumeRolePolicy.Statement &&
+                assumeRolePolicy.Statement.length &&
+                assumeRolePolicy.Statement[0].Principal &&
+                assumeRolePolicy.Statement[0].Principal.AWS) {
```

Comment:
We need to check all of the statements, not just `[0]`, so this should be a loop for this section.

Reply:
So I created a role and as per my understanding, these settings are role-based. I couldn't find a scenario where there will be more than once statements against one role policy document.
![Capture](https://user-images.githubusercontent.com/31914988/91971619-f010c500-ed32-11ea-8137-03e91d7ae496.PNG)

Reply:
And if there are multiple statements, should we check that at least one should have MFA/external ID enabled or all statements?

Reply:
Please check that each statement has either mfa or external ID enabled